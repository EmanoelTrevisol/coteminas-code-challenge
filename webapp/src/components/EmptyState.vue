<template>
  <p>{{ title }}</p>
</template>

<script>
export default {
  props: {
    title: {
      type: String,
      default: 'Nenhum resultado foi encontrado',
    },
  },
};
</script>
