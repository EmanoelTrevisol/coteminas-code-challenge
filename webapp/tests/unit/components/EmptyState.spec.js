import { shallowMount } from '@vue/test-utils';
import EmptyState from '@/components/EmptyState';

describe('EmptyState.vue', () => {
  it('renders EmptyState component', () => {
    const wrapper = shallowMount(EmptyState);

    expect(wrapper.html()).toMatchSnapshot();
  });

  it('renders default text correctly', () => {
    const text = 'Nenhum resultado para sua pesquisa foi encontrado';

    const wrapper = shallowMount(EmptyState, {
      propsData: {
        text,
      },
    });

    const textWrapper = wrapper.find('.text');

    expect(textWrapper.exists()).toBe(true);
    expect(textWrapper.text()).toBe(text);
  });

  it('renders prop text correctly', () => {
    const text = 'Nada encontrado!';

    const wrapper = shallowMount(EmptyState, {
      propsData: {
        text,
      },
    });

    const textWrapper = wrapper.find('.text');

    expect(textWrapper.exists()).toBe(true);
    expect(textWrapper.text()).toBe(text);
  });
});
