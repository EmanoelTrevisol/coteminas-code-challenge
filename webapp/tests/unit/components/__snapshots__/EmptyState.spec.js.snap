// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EmptyState.vue renders EmptyState component 1`] = `
<div class="empty-state">
  <h3>Ops... n√£o tem nada por aqui</h3>
  <p class="text">Nenhum resultado para sua pesquisa foi encontrado</p>
</div>
`;
